examples/docker/03_dind_feature/Dockerfile

```diff
@@ -1,3 +1,18 @@
 FROM ubuntu:noble
-ADD entrypoint.sh /entrypoint.sh
-ENTRYPOINT ["/entrypoint.sh"]
\ No newline at end of file
+# Install some dependencies such as curl and sudo.
+# Also set up passwordless sudo for the ubuntu user.
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -y \
+      curl \
+      sudo \
+      apt-transport-https && \
+    echo "ubuntu ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/ubuntu
+# Add our onCreateCommand script.
+ADD on-create.sh /on-create.sh
+# Switch to the non-root user.
+USER ubuntu
+# The devcontainer feature provides /usr/local/share/docker-init.sh
+# which will handle most of the steps of setting up Docker.
+# We can't put this in the entrypoint as it gets overridden, so
+# we call it in the on-create script.
```

Comment:
Do we not support setting entrypoint in the Dockerfile at all? Maybe we should in the future.

Reply:
My observation is that whatever entrypoint you set gets overridden by `ENVBUILDER_INIT_COMMAND` / `ENVBUILDER_INIT_SCRIPT`.

Reply:
For a Coder workspace we need to have `coder_agent.*.init_script` as the entry point so another option to handle the `ENTRYPOINT` is to run it as part of `coder_agent` `startup_script`. This will work for at least the case when envbuilder is used with Coder.

Reply:
Filed https://github.com/coder/envbuilder/issues/351 to follow up on this.